src/snapred/backend/service/CalibrationService.py

```diff
@@ -74,7 +75,8 @@ def __init__(self):
         super().__init__()
         self.dataFactoryService = DataFactoryService()
         self.dataExportService = DataExportService()
-        self.groceryService = FetchGroceriesRecipe()  # TODO replace with service
+        self.groceryService = GroceryService()
+        self.groceryClerk = GroceryListItem.builder()
```

Comment:
Does clerk ever get reset or is it just building a bigger and bigger list of groceries every call?

Reply:
I see that it does reset [here](https://github.com/neutrons/SNAPRed/pull/177/files#diff-94ad68da3b204ecedce9e4af74b8fcb7cfbfb4e8f48d35f0bcc0e206eea4ad24R95), in my experience `build()` is terminal in the lifespan of a builder, youd just make a new one if you wanted to make a new object.

Reply:
The builder has the ability to create not just single items but also entire lists, and that is what leads it to this behavior.

The builder tracks two things, `_tokens` and `_list`.  `_tokens` is for building individual list items, and `_list` is for building an entire list.

The user can keep adding item specifications, then call either `build()` or `add()` to create the item, which also clears `_tokens`.  So a single builder can safely make an entire list of items using the `add()` method.  So the builder needs to be reusable.

If they use `add()`, then later a command `buildList()` or `buildDict()` will return a list of all the items made, and in the process clears `_list`.   It has to clear `_list`, or else the user could do the thing you're describing, with unintentional ever-increasing grocery lists of workspaces.

Because it has to be safely reusable, it might as well be reused.  That was my thought, anyway.